contents/tutorials/chatgpt-analytics.md

```diff
@@ -0,0 +1,318 @@
+- `total_cost_in_dollars` i.e. `input_cost_in_dollars + output_cost_in_dollars`
+
+
+Update your `fetchChatGPTResponse()` function in `App.js` to capture this event:
+
+```js file=App.js
+const fetchChatGPTResponse = async () => {
+    try {
+
+      // your existing code...
+
+      const chatCompletion = await openai.chat.completions.create({
+        messages: [{ 
+          role: "user",
+          content: userInput }],
+        model: selectedModel.name,
+      });
+      posthog.capture('chat_completion', {
+        model: chatCompletion.model,
+        prompt: userInput,
+        prompt_tokens: chatCompletion.usage.prompt_tokens,
+        completion_tokens: chatCompletion.usage.completion_tokens,
+        total_tokens: chatCompletion.usage.total_tokens,
+        input_cost_in_dollars: chatCompletion.usage.prompt_tokens * selectedModel.token_input_cost,
+        output_cost_in_dollars: chatCompletion.usage.prompt_tokens * selectedModel.token_output_cost,
+        total_cost_in_dollars: inputCostInDollars + outputCostInDollars
+      })
+
+      // your existing code...
+```
+
+Refresh your app and submit a few prompts. You should then see your events captured in the [PostHog Activity tab](https://us.posthog.com/events).
+
+<ProductScreenshot
+  imageLight={EventsLight} 
+  imageDark={EventsDark} 
+  alt="ChatGPT completion events in PostHog" 
+  classes="rounded"
+/>
+
+## 3. Create insights
+
+Now that we're capturing events, we can create [insights](/docs/product-analytics/insights). Below are three examples of useful metrics you should monitor:
+
+### Total cost
+
+To create this insight, go the [Product analytics tab](https://us.posthog.com/insights) and click **+ New insight**. Then:
+
+1. Set the event to `chat_completion`
+2. Click on **Total count** to show a dropdown. Click on **Property value (sum)**.
+3. Select the `total_cost_in_dollars` property.
+
+Then, change the chart type from **Line chart** to **Number** (or however else you'd like to visualize your data). Note that it may show `0` if your total cost is smaller than `0.01`.
+
+Additionally, you can also breakdown your costs by model. To do this, click **+ Add breakdown** and select `model` from the event properties list.
+
+<ProductScreenshot
+  imageLight={TotalCostLight} 
+  imageDark={TotalCostDark} 
+  alt="Total ChatGPT model costs in PostHog" 
+  classes="rounded"
+/>
+
+### Average cost per user
+
+This metric helps give you an idea of how your costs will scale as your product grows. Creating this insight is similar to creating the one above, however we use **formula mode** to divide the total cost by the total number of users:
+
+1. Set the event to `chat_completion`
+2. Click on **Total count** to show a dropdown. Click on **Property value (sum)**.
+3. Select the `total_cost_in_dollars` property.
+4. Click **+ Add graph series**.
+5. Keep the event name as `All events`, but change the value from `Total count` to `Unique users`.
```

Comment:
Wouldn't you only care about users of the LLM feature? Doesn't matter that much though.